### ISSUE TITLE
Migrate all pipelines off Ubuntu 18.04 by 4/1/2023

### ISSUE THREAD

**Issue by mikeharder** [Azure/azure-sdk-tools#5472]
The Ubuntu 18.04 image will be fully unsupported on 4/1/2023 (actions/runner-images#5583).  Any pipelines still using Ubuntu 18.04 should be upgraded to Ubuntu 20.04 or 22.04 before this date.

You can view the pipelines still running in pool `azsdk-pool-mms-ubuntu-1804-general` here:

https://dev.azure.com/azure-sdk/internal/_settings/agentqueues?queueId=120&view=jobs

I found the following repos still using Ubuntu 18.04.  Search for "18.04" or "1804".

https://github.com/Azure/azure-rest-api-specs-pipeline
https://github.com/azure/azure-sdk-for-c
https://github.com/azure/azure-sdk-for-cpp
https://github.com/Azure/azure-sdk-for-python

Related work:
- https://github.com/Azure/azure-sdk-tools/issues/3407#issuecomment-1454165893
- https://github.com/Azure/azure-sdk-tools/issues/5494

Page to check which builds are still using the Ubuntu 18 pool:
https://dev.azure.com/azure-sdk/_settings/agentpools?poolId=19&view=jobs

**Comment by konrad-jamrozik** [comment]
PR:
- https://devdiv.visualstudio.com/DevDiv/_git/openapi-alps/pullrequest/455967

**Comment by konrad-jamrozik** [comment]
Related issue, possible blocker:

- https://github.com/Azure/azure-sdk-for-net/issues/31839

There is also relevant email thread about it: 
> RE: Error of "No usable version of the libssl was found" when running autorest.csharp V2 in swagger pipeline

**Comment by konrad-jamrozik** [comment]
Search query:
https://github.com/search?q=org%3AAzure+%28path%3A*.y*ml+OR+path%3A*.json%29+MMSUbuntu18&type=code

**Comment by mikeharder** [comment]
Additional searches that may catch a few more cases:

**Name of our DevOps Pool**
https://github.com/search?q=org%3AAzure+%28path%3A*.y*ml+OR+path%3A*.json%29+azsdk-pool-mms-ubuntu-1804-general&type=code

**Name of GitHub Actions Image (needs to be filtered to repos we own)**
https://github.com/search?q=org%3AAzure+%28path%3A*.y*ml+OR+path%3A*.json%29+ubuntu-18.04&type=code

**Comment by konrad-jamrozik** [comment]
All the queries merged into one:

`org:Azure (path:*.y*ml OR path:*.json) MMSUbuntu18 OR ubuntu18 OR azsdk-pool-mms-ubuntu-1804-general OR ubuntu-18.04`  
https://github.com/search?q=org%3AAzure+%28path%3A*.y*ml+OR+path%3A*.json%29+MMSUbuntu18+OR+ubuntu18+OR+azsdk-pool-mms-ubuntu-1804-general+OR+ubuntu-18.04&type=code

**Comment by konrad-jamrozik** [comment]
I am not submitting any PR against https://github.com/Azure/embedded-wireless-framework/ because:

- It doesn't appear to use [the Ubuntu 18 pool](https://dev.azure.com/azure-sdk/_settings/agentpools?poolId=19&view=jobs)
- All occurrences of `azsdk-pool-mms-ubuntu-1804` in the repository are either in docs or `examples` dirs.

**Comment by konrad-jamrozik** [comment]
For https://github.com/Azure/azure-uamqp-python I made a direct commit, a no PR was necessary:
https://github.com/Azure/azure-uamqp-python/commit/bf1118c156bcf863250ef87f26bc29159d7f1674

**Comment by konrad-jamrozik** [comment]
I am not submitting any PR against https://github.com/Azure/iotedgedev

The [search query](https://github.com/search?q=repo%3AAzure%2Fiotedgedev+%28azsdk-pool-mms-ubuntu-1804+OR+MMSUbuntu18.04%29&type=code) returned only this:

```
pool:
  name: Azure-IoT-EdgeExperience-1ES-Hosted-Linux
  demands:
    - ImageOverride -equals MMSUbuntu18.04TLS
```

which is unrelated to our pools or images.

**Comment by konrad-jamrozik** [comment]
Status update as of 4/21/2023 PDT 6:30 PM. I did all the necessary updates.

Once this PR is merged:
- https://github.com/Azure/azure-sdk-for-c/pull/2549

and search index updated to current `main`, then:

- [this query](https://github.com/search?q=org%3AAzure+%28path%3A*.y*ml+OR+path%3A*.json%29+%28%28MMSUbuntu18+OR+azsdk-pool-mms-ubuntu-1804%29+AND+NOT+MMSUbuntu18.04TLS%29+-repo%3AAzure%2Fembedded-wireless-framework&type=code
) 
- We should no longer see any use of the [`
azsdk-pool-mms-ubuntu-1804-general` pool](https://dev.azure.com/azure-sdk/_settings/agentpools?poolId=19&view=jobs)

-

**Comment by konrad-jamrozik** [comment]
I got a notification of [following failure](https://github.com/Azure/azure-uamqp-python/actions/runs/4769673479) for https://github.com/Azure/azure-uamqp-python:

![image](https://user-images.githubusercontent.com/4429827/234078820-e92f3eda-f4e3-41a8-9123-c7953754f880.png)

Attempting fix by following commit:
- https://github.com/Azure/azure-uamqp-python/commit/bf004b01db65e18b26d194d55335578369179d70

**Comment by konrad-jamrozik** [comment]
There have been no new builds scheduled since April 24:
https://dev.azure.com/azure-sdk/_settings/agentpools?poolId=19&view=jobs

Closing this issue as resolved.
